### DISCUSSION TITLE
Having difficulty working with forge on windows 10

### DISCUSSION THREAD

**Discussion by freeman237** [Cyfrin/foundry-full-course-cu#2404] [Q&A]
I cannot check the forge version on vscode but i can on the ubuntu app. On vscode i get the following error
![forge](https://github.com/user-attachments/assets/8655b15b-974e-400c-a077-ef6cae389b77)

**Comment by freeman237** [comment]
![Screenshot 2024-09-05 220801](https://github.com/user-attachments/assets/27b75f01-0890-408e-b12d-205e53777d60)

**Reply by EngrPips** [reply]
Hello. Have you installed the `foundry` yet? if not, follow this installation guide https://book.getfoundry.sh/getting-started/installation  to install Foundry.

**Reply by freeman237** [reply]
Thank you Sir, but its still not working
I still get the same error on vs code. It seems OK on the ubuntu software but on vs code it does not work

**Reply by EngrPips** [reply]
You are welcome. I use Mac, and I am not sure how Ubuntu software works, but let's hope someone who knows sees this and helps. We will appreciate any help @cromewar.

**Comment by Haardtripathi** [comment]
You might need to install WSL on your laptop, and use VS code on that WSL for this course.
If you need the steps for installing and integrating WSL feel free to ask.

**Reply by freeman237** [reply]
Please i need the steps to intergrate the wsl, I have installed it already

**Reply by Haardtripathi** [reply]
Ok then,
1.open up microsoft store and install Ubuntu.
2.install remote wsl extension in vsCode
3.open up Ubuntu terminal, and run the following commands =>
wsl --user root

passwd username
Replace username with your actual Linux username.
You will be asked to type a new password without being asked for the current one

4.setup git in Ubuntu terminal 
sudo apt update
sudo apt install git -y

git config --global user.name "Your Name"
Replace "Your Name" with your actual name

git config --global user.email "youremail@example.com"
Replace "youremail@example.com"

5.openup VS code
Ctrl+Shift+P
You will get option for opening WSL:Connect to WSL
![image](https://github.com/user-attachments/assets/fc0ac9a4-8567-4133-b64c-bafb30cfac88)

**Reply by Haardtripathi** [reply]
@freeman237 Tell me if you face any issue during the process.

**Reply by freeman237** [reply]
Thank you  @Haardtripathi, I still get error
I have even tried using the link to download the missing file but i get not found
![Screenshot 2024-09-07 091040](https://github.com/user-attachments/assets/9f78f805-6ac4-4d41-9606-75007639729c)

**Reply by Haardtripathi** [reply]
try this https://youtu.be/6VMQMoVYfo8?si=Gs8smc9eIOqxe32I

**Comment by EngrPips** [comment]
Interesting

**Comment by TrustWarden** [comment]
Install WSL extension to connect to the linux subsystem and also use bash instead of powershell, if it isn't shown in your vscode then install git, it will fix it

**Comment by EngrPips** [comment]
Thanks for the help @Haardtripathi

**Reply by Haardtripathi** [reply]
Welcome brother. I really appreciate your work in the discussions.

**Reply by EngrPips** [reply]
I am grateful for the commendation, sir.
